app/controllers/employments/AddEmploymentController.scala

```diff
@@ -79,236 +69,245 @@ class AddEmploymentController @Inject()(personService: PersonService,
-    implicit person =>
-      implicit request =>
-        journeyCacheService.currentCache map {
-          currentCache =>
-            EmploymentAddDateForm(currentCache(AddEmployment_NameKey)).form.bindFromRequest().fold(
-              formWithErrors => {
-                BadRequest(views.html.employments.add_employment_start_date_form(formWithErrors, currentCache(AddEmployment_NameKey)))
-              },
-              date => {
-                val startDateBoundary = new LocalDate().minusWeeks(6)
-                val data = currentCache + (AddEmployment_StartDateKey -> date.toString)
-                if (date.isAfter(startDateBoundary)) {
-                  val firstPayChoiceCacheData = data + (AddEmployment_StartDateWithinSixWeeks -> YesValue)
-                  journeyCacheService.cache(firstPayChoiceCacheData)
-                  Redirect(controllers.employments.routes.AddEmploymentController.receivedFirstPay())
-                } else {
-                  val firstPayChoiceCacheData = data + (AddEmployment_StartDateWithinSixWeeks -> NoValue)
-                  journeyCacheService.cache(firstPayChoiceCacheData)
-                  Redirect(controllers.employments.routes.AddEmploymentController.addEmploymentPayrollNumber())
-                }
-              }
-            )
+      EmploymentNameForm.form.bindFromRequest.fold(
+        formWithErrors => {
+          Future.successful(BadRequest(views.html.employments.add_employment_name_form(formWithErrors)))
+        },
+        employmentName => {
+          journeyCacheService.cache(Map(AddEmployment_NameKey -> employmentName))
+            .map(_ => Redirect(controllers.employments.routes.AddEmploymentController.addEmploymentStartDate()))
         }
+      )
   }
 
-  def receivedFirstPay(): Action[AnyContent] = authorisedForTai(personService).async { implicit user =>
-    implicit person =>
-      implicit request =>
-        ServiceCheckLite.personDetailsCheck {
-          journeyCacheService.collectedValues(Seq(AddEmployment_NameKey), Seq(AddEmployment_RecewivedFirstPayKey)) map tupled { (mandSeq, optSeq)  =>
-            Ok(views.html.employments.add_employment_first_pay_form(AddEmploymentFirstPayForm.form.fill(optSeq(0)), mandSeq(0)))
-          }
+  def addEmploymentStartDate(): Action[AnyContent] = (authenticate andThen validatePerson).async {
+    implicit request =>
+
+      implicit val user = request.taiUser
+
+      (journeyCacheService.collectedValues(Seq(AddEmployment_NameKey), Seq(AddEmployment_StartDateKey)) map tupled { (mandSeq, optSeq) =>
+
+        val form = optSeq(0) match {
+          case Some(dateString) => EmploymentAddDateForm(mandSeq(0)).form.fill(new LocalDate(dateString))
+          case _ => EmploymentAddDateForm(mandSeq(0)).form
         }
+        Ok(views.html.employments.add_employment_start_date_form(form, mandSeq(0)))
+      }).recover {
+        case _ => internalServerError("Could not add employment start date")
```

Comment:
could recover the exception here with:

case NonFatal(e) => internalServerError("Could not add employment start date", Some(e))